File: 백준/Silver/27496. 발머의 피크 이론/발머의 피크 이론.py

n, l = map(int,input().split())
a = list(map(int,input().split())) 

current = 0
cnt = 0

for i in range(n):
    current += a[i]
    if i >= l:
        current -= a[i - l]
    if 129 <= current <= 138:
        cnt += 1
        
print(cnt)

File: 백준/Silver/27496. 발머의 피크 이론/발머의 피크 이론.js
const input = require("fs").readFileSync("/dev/stdin", "utf-8").trim().split("\n");

const [n, l] = input[0].trim().split(" ").map(Number);
const a = input[1].trim().split(" ").map(Number);

let current = 0;
let cnt = 0;

for (let i = 0; i < n; i++) {
	current += a[i];
	if (i >= l) current -= a[i - l];
	if (current >= 129 && current <= 138) cnt++;
}
console.log(cnt);
